Overhauled the Edit Service and the Edit User

rkallby opened the pull request:
Significantly cleaned up the edit user and edit service forms. Also have fixed some of the logic protecting against invalid entries. Also updated the success popup form which was originally just a 
form with a label

jhinkle97 commented on ChocAnonGUI/GUI/Operator/EditServiceForm.cs:
```diff
@@ -11,76 +12,47 @@
-        private string procedure = "";
-     
-        
+        private string serviceCode;
```
Can we make this a local variable in searchButton_Click and just pass service.Code in the delete function?

rkallby replied:
I'd prefer not as it adds another layer of complexity. This way it's easier to access in both the delete and edit functions. I can though if you really want me too.

rkallby replied:
Left that as is.

jhinkle97 replied:
We already have a global service variable that contains this value though, right?

rkallby replied:
Ah, nevermind... I see what you mean. Yeah, I can just switch out serviceCode and make it local. Then inn deleteButton I cann make it service.code. My mistake.

rkallby replied:
I'll change it rn

rkallby replied:
changes have been made in the latest commit

jhinkle97 commented on ChocAnonGUI/GUI/Operator/EdituserForm.cs:
```diff
@@ -49,67 +50,90 @@ private void searchButton_Click(object sender, EventArgs e)
-            zipTextbox.Text = user.Zip;
-
-            if (nameTextbox.Text == "")
+            if (user.UserNumber == "" || user.Name == "")
```
Any reason we are checking the name as well?

rkallby replied:
It was having issues with some accounts if I just used the User Number.

rkallby replied:
Reverted back to just the user.usernumber. If we find an issue doing final testing tomorrow we can patch it up.

jhinkle97 replied:
Okay, I think that if the get user function finds a user, then the user number will never be an empty string unless there is a user in the dB who’s number is an empty string and u search using an empty string. This shouldn’t ever happen though since we generate the numbers when creating users.
Also, if the get user function does not find a user it will return a new user object every time where the user number will be an empty string.

jhinkle97 commented on ChocAnonGUI/GUI/Operator/EdituserForm.cs:
```diff
@@ -49,67 +50,90 @@ private void searchButton_Click(object sender, EventArgs e)
-            cityTextbox.ReadOnly    = false;
-            addressTextbox.ReadOnly = false;
-            zipTextbox.ReadOnly     = false;
+            if (nameTextbox.Text == "" || enableButtons == false || user.UserNumber == "")
```
Same as previous comment

rkallby replied:
Just to be extra thorough in the case the user.usercode didn't work.

rkallby replied:
Reverted it back to user.userNumber as per above

Commit:
Overhauled the Edit Service and Edit User forms. Still need to make a success popup for new Services taht are added/edited.

Commit:
Made EditService Fixed Window

Commit:
Made success form fixed size and populate in the center of the screen

Commit:
Set the char limit to 6 on the service edit entry
set the char limit to the edit service name to 20
set the char limit to the edit service fee to 6
set the char limit to search a user to 9 chars.

Commit:
Made it so if a member is suspended, their suspended status will show up red.

Commit:
Changes made with naming and a popup for suspended member;

Commit:
Changed Error to Warning

jhinkle97 reviewed (approved):
Looks good, thanks!

rkallby commented:
No problem, thank you for the tips and pointers, if you haven't gotten the
idea already, I'm definitely not the most precise/sharpened programer and I
sincerely appreciate the feedback !! :)

On Tue, May 5, 2020, 8:40 PM jhinkle97 <notifications@github.com> wrote:

> *@jhinkle97* approved this pull request.
>
> Looks good, thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jhinkle97/software-engineering/pull/29#pullrequestreview-406245646>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBLS532J6YW4B455S3GTI3RQCWYNANCNFSM4MZ7AYOQ>
> .
>

Commit:
Made changes to fix the if conditions based on the comments

Commit:
Changed the serviceCode to a local variable and then just pulled service.Code in the deleteButton function
